src/service/data_products/common_models.py

```diff
@@ -77,11 +82,9 @@ def _check_router_tags(cls, v):  # @NoSelf
     @field_validator("db_collections")
     @classmethod
     def _ensure_only_one_view(cls, v):
-        found = False
-        for dbc in v:
-            if found and v.view_required:
-                raise ValueError("More than one db collection requiring a view found")
-            found = found or dbc.view_required
+        view_required_collections = [dbc for dbc in v if dbc.view_required or dbc.generic_view_required]
+        if len(view_required_collections) > 1:
+            raise ValueError("More than one db collection requiring a view found")
         return v
     
     def view_required(self):
```

Comment:
Can you update the method docs to say
```
Check if a non-generic search view is required for this data product. 
```
Unfortunately you can't make a suggestion on a hidden line

Reply:
👍

Reply:
:+1: